### ISSUE TITLE
GitHub Actions ?

### ISSUE THREAD

**Issue by MSP-Greg** [eventmachine/eventmachine#899]
At present, GitHub Actions allows testing on Ubuntu 18.04, macOS, & Windows for Ruby versions 2.4 thru 2.6.  Using Ubuntu 16.04, Ruby 2.3 can be tested.

Ubuntu & Windows are passing, macOS seems to have issues, some time related.

Interested?  See https://github.com/MSP-Greg/eventmachine/runs/218249167

**Comment by sodabrew** [comment]
Yes, yes I absolutely do want to switch over to GitHub Actions especially
to generate binaries!

>

**Comment by MSP-Greg** [comment]
@sodabrew

Sorry for disappearing.  I can push this, but the macOS tests all failed, seems to be 1 error and 2 or 3 failures, so nothing extreme.

1. If we want to continue to test on various OS/OpenSSL, it might be a good idea to leave some Travis jobs.

2. The Windows part of the workflow does depend on a 'GitHub Action' that's one of my repos.  It's part of the 'GitHub Action' system (independent actions), so nothing needs to be added here.

As a reminder, the Windows builds use the MSYS2 build tools, and the action allows one to easily update the tools, choose the correct OpenSSL version, etc.  For instance, on Windows, Ruby 2.4.x uses OpenSSL 1.0.2, later versions all use 1.1.1.  Some repos require ragel, which isn't installed, etc...

> to generate binaries!

Which ones?
